activerecord/lib/active_record/database_configurations.rb

```diff
@@ -79,10 +79,12 @@ def default_hash(env = default_env)
     # If the application has multiple databases +find_db_config+ will return
     # the first DatabaseConfig for the environment.
     def find_db_config(env)
-      configurations.find do |db_config|
-        db_config.env_name == env.to_s ||
-          (db_config.for_current_env? && db_config.name == env.to_s)
-      end
+      configurations
+        .sort_by { |db_config| db_config.for_current_env? ? 0 : 1 }
```

Comment:
I'm disregarding the fact that we're doing two loops here since the size of the configurations array should be fairly small.